### ISSUE TITLE
Tests aborted with "malloc(): corrupted top size"

### ISSUE THREAD

**Issue by david-prv** [Jamesbarford/holyc-lang#43]
When running `make unit-test` in `/src` folder, it yields:
```bash
$ make unit-test 
cd ../tests && hcc -o test-runner ./run.HC && ./test-runner
malloc(): corrupted top size
Aborted (core dumped)
make: *** [Makefile:53: unit-test] Error 134
```

**Comment by Jamesbarford** [comment]
Thanks for this, I'll need to dig into these errors you've found. Which operating system are you on?

**Comment by david-prv** [comment]
Hi @Jamesbarford, thank you! I'm on `Linux Mint 21.2 Cinnamon`

**Comment by Jamesbarford** [comment]
Thank you for finding these, very helpful. Both the issue of `hcc -tokens` and `make unit-test` should now be resolved on the `main` branch. (fixed by #44)
